pulsar-io/kafka-connect-adaptor/src/test/java/org/apache/pulsar/io/kafka/connect/KafkaConnectSourceTest.java

```diff
@@ -101,6 +104,132 @@ public void testOpenAndReadTaskDirect() throws Exception {
         testOpenAndReadTask(config);
     }
 
+    @Test
+    void testCastTransformation() throws Exception{
```

Comment:
There is a fair amount of duplication in these tests. I will address this shortly.